config/routes.rb

```diff
@@ -1,19 +1,27 @@
 Rails.application.routes.draw do
     get 'homepage/index'
 
+    # Login/OAuth routes
     get "/login", to: redirect("/auth/google_oauth2")
     get "/auth/google_oauth2/callback", to: "sessions#create"
     delete "/logout", to: "sessions#destroy"
     get "/logout", to: "sessions#destroy"
 
-    post   '/users/:user_id/courses/:id', to: 'courses#edit_flashcard_order'
+    # Student Roster routes
     get    '/users/:user_id/courses/:id/student_index', to: 'courses#student_index'
+
+    # Flashcard routes
     get    '/users/:user_id/courses/:id/flashcard',  to: 'courses#flashcard'
     get    '/users/:user_id/courses/:id/quiz',  to: 'courses#quiz'
     post   '/users/:user_id/courses/:id/update_notes',  to: 'courses#updateNotes'
+    post   '/users/:user_id/courses/:id', to: 'courses#edit_flashcard_order'
 
-    get    '/:assignment_id/search',  to: 'submissions#search'
 
+    # Poll routes
+    get    '/poll/:poll_id', to: 'polls#student_show'
```

Comment:
The /poll url is a good 1.0 fix, but I think we should go back to this later and see if there's a better option

Reply:
I agree--Mary Clare had a better idea about how to handle this, but it will take a bit of work. It's probably more a phase 4 thing.